engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/execution/CommandExecutionEngine.scala

```diff
@@ -59,8 +76,23 @@ class CommandExecutionEngine(interpreterContext: InterpreterContext)
     )
 
   /** @inheritdoc */
-  def invoke(cmd: Command): cmd.Result[Completion] = {
-    cmd.execute(runtimeContext, commandExecutionContext)
+  def invoke(cmd: Command): Future[Completion] = {
+    cmd match {
+      case c: SynchronousCommand =>
+        blockingQueue.add(c)
+        val javaFuture =
+          sequentialExecutionService.submit(new Callable[Completion] {
+            override def call(): Completion = {
+              val orderedCommand = blockingQueue.take()
+              orderedCommand.execute(runtimeContext, sequentialExecutionContext)
+            }
+          })
+        Future {
+          javaFuture.get()
```

Comment:
Creating new future that will block a thread and wait for Java future!? We don't have _virtual threads_ yet and regular threads take significant system resources.

Why not `javaFuture.asScala` as https://stackoverflow.com/questions/35033290/convert-a-java-future-to-a-scala-future suggests?

Reply:
`asScala` is an implicit on `CompleteableFuture` not `Future` so that won't work out of the box. But let me tinker with it.